packages/react/src/components/Link/Link.tsx

```diff
@@ -88,6 +89,7 @@ const LinkBase = React.forwardRef<
   (
     {
       as: BaseComponent,
+      asWithCarbonClasses = false,
```

Comment:
default is `false` to maintain current behavior (will not affect current use-cases)